website/addons/evernote/views.py

```diff
@@ -0,0 +1,215 @@
+    client = utils.get_evernote_client(token)
+
+    # will want to pick up notes for the notebook
+    # start with calculating the number of notes in nb
+
+    notes = utils.notes_metadata(client,
+                    notebookGuid=node_addon.folder_id,
+                    includeTitle=True,
+                    includeUpdated=True,
+                    includeCreated=True)
+
+    results = [{'title': note.title,
+              'guid': note.guid,
+              'created': utils.timestamp_iso(note.created),
+              'updated': utils.timestamp_iso(note.updated)}
+              for note in notes]
+
+    return results
+
+@must_have_addon('evernote', 'node')
+@must_be_addon_authorizer('evernote')
+def evernote_note(node_addon, noteid, **kwargs):
+
+    token = node_addon.external_account.oauth_key
+    client = utils.get_evernote_client(token)
+    note = utils.get_note(client, noteid,
+        withContent=True,
+        withResourcesData=False)
+
+    # convert to HTML
+    mediaStore = utils.MyMediaStore(client.get_note_store(), note.guid)
+    html = ENML2HTML.ENMLToHTML(note.content, pretty=True, header=False,
+          media_store=mediaStore)
+
+    return {'noteid': noteid,
+            'content': note.content,
+            'html': html}
+
+@must_have_addon('evernote', 'node')
+@must_be_addon_authorizer('evernote')
+def evernote_folder_list(node_addon, **kwargs):
+    """Returns a list of notebooks in Evernote"""
+    if not node_addon.has_auth:
+        raise HTTPError(http.FORBIDDEN)
+
+    node = node_addon.owner
+    folder_id = request.args.get('notebookId')
+
+    # now grab the notebook list
+
+    # figure out how to recast into the format that the folderpicker likes
+
+    if folder_id is None:
+        return [{
+            'id': '0',
+            'path': 'All Notes',
+            'addon': 'evernote',
+            'kind': 'folder',
+            'name': '/ (Full Evernote)',
+            'urls': {
+                'folders': node.api_url_for('evernote_folder_list', notebookId='0'),
+            }
+        }]
+    elif folder_id == '0':
+
+        # return all the notebooks
+        # stacks can contain noteboooks but not other stacks
+        # first pass: ignore stacks
+
+        # if I incorporate stacks, any setting to not allow it to be selected.
+
+        token = node_addon.external_account.oauth_key
+        client = utils.get_evernote_client(token)
+        notebooks = utils.get_notebooks(client)
+
+        return [{
+            'id': notebook['guid'],
+            'path': notebook['name'],
+            'addon': 'evernote',
+            'kind': 'folder',
+            'name': notebook['name'],
+            'urls': {
+                'folders': node.api_url_for('evernote_folder_list', notebookId=notebook['guid']),
+            }
+
+        } for notebook in notebooks]
+    else:
+        return []
+
+@must_be_contributor_or_public
+@must_have_addon('evernote', 'node')
+def evernote_widget(node_addon, **kwargs):
```

Comment:
This widget will likely be going away, but possibly replaced with a settings widget to select the Notebook. It should be possible to use FolderPicker for that, but it might be simpler to use a widget.